pkg/archive/jar_test.go

```diff
@@ -176,3 +177,122 @@ func TestStableJARBuildMetadata(t *testing.T) {
 		})
 	}
 }
+
+func TestStableOrderOfAttributeValues(t *testing.T) {
+	testCases := []struct {
+		test          string
+		attributeName []string
+		input         []*ZipEntry
+		expected      []*ZipEntry
+	}{
+		{
+			test:          "single_attribute",
+			attributeName: []string{"Provide-Capability"},
+			input: []*ZipEntry{
+				{
+					&zip.FileHeader{Name: "META-INF/MANIFEST.MF"},
+					[]byte("Provide-Capability: sling.servlet;sling.servlet.resourceTypes:List<Strin\n g>=\"org/apache/sling/scripting/sightly/testing/precompiled\";scriptEngin\n e=rhino;scriptExtension=ecma;sling.servlet.selectors:List<String>=scrip\n t,sling.servlet;sling.servlet.resourceTypes:List<String>=\"org/apache/sl\n ing/scripting/sightly/testing/precompiled\";scriptEngine=rhino;scriptExt\n ension=js;sling.servlet.selectors:List<String>=script,sling.servlet;sli\n ng.servlet.resourceTypes:List<String>=\"org/apache/sling/scripting/sight\n ly/testing/precompiled\";scriptEngine=htl;scriptExtension=html,sling.ser\n vlet;sling.servlet.resourceTypes:List<String>=\"org/apache/sling/scripti\n ng/sightly/testing/precompiled/templates-access-control\";scriptEngine=h\n tl;scriptExtension=html,sling.servlet;sling.servlet.resourceTypes:List<\n String>=\"org/apache/sling/scripting/sightly/testing/precompiled/templat\n es-access-control\";scriptEngine=htl;scriptExtension=html;sling.servlet.\n selectors:List<String>=\"partials,include\"\n"),
```

Comment:
would you be able to add a smaller, synthetic example, as well, that has a super simple case e.g. "c,b,a" -> "a,b,c"? the realistic cases are nice but only insofar as it's clear they're exercising some aspect of the implementation.

Reply:
[`c2ce156` (#339)](https://github.com/google/oss-rebuild/pull/339/commits/c2ce1565bc9a0d95a632c750d01d1178ce7bb017)